pysmt/smtlib/script.py

```diff
@@ -77,16 +77,53 @@ def serialize(self, outstream=None, printer=None, daggify=True):
             printer.printer(self.args[0])
             outstream.write(")")
 
+        elif self.name == smtcmd.ASSERT_SOFT:
+            outstream.write("(%s " % self.name)
+            printer.printer(self.args[0])
+            for arg in self.args[1]:
+                option_name, value = arg
+                if option_name == ":weight":
+                    outstream.write(" %s " % option_name)
+                    printer.printer(value)
+                else:
+                    outstream.write(" %s %s" % (option_name, value))
+            outstream.write(")")
+
+
         elif self.name == smtcmd.GET_VALUE:
             outstream.write("(%s (" % self.name)
             for a in self.args:
                 printer.printer(a)
                 outstream.write(" ")
             outstream.write("))")
 
+        elif self.name in [smtcmd.MAXIMIZE, smtcmd.MINIMIZE]:
+            outstream.write("(%s " % self.name)
+            printer.printer(self.args[0])
+            for arg in self.args[1]:
+                if len(arg) == 1:
+                    outstream.write(" %s" % arg[0])
+                elif len(arg) == 2:
+                    outstream.write(" %s " % arg[0])
+                    printer.printer(arg[1])
+                elif len(arg) > 2:
```

Comment:
I would not try to enforce the shape of this command here. We should rely on the fact that commands are well-formed, and not restrict what the user might want to do here.

I do not know how the rest of the code was using this, so there might be a reason for this.